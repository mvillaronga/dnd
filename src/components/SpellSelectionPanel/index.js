import React from 'react'
import { useSelector } from 'react-redux'

import SpellFilter from '../SpellFilter'
import SpellList from '../SpellList'

const SpellSelectionPanel = () => {
    
    const props = {
        spells: useSelector(state => state.refs.spells.items)
    }

    return (
    <div>
        <h1>Spells</h1>
        <hr />
        <SpellFilter />
        <hr />
        <SpellList {...props} />
    </div>
)}

export default SpellSelectionPanel