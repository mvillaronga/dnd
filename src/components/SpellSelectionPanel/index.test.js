import React from 'react'
import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import SpellSelectionPanel from '../SpellSelectionPanel'

configure({ adapter: new Adapter() });

const spell = {name: "friends", level:"0"} //odd selector issues in the state store
const state = {
  selection: {
    spells: [
      spell,
    ]
  },
  refs: {
    spells: {
      items: [
        {name: "magic missle", level:"1"},
        {name: "find familliar", level:"1"},
        {name: "fire bolt", level:"0"},
        spell
      ]
    }
  }
}

const mockStore = configureMockStore([thunk])
const store = mockStore(state)
const wrapper = () => mount(
  <Provider store={store}>
    <SpellSelectionPanel />
  </Provider>
)

describe('NpcListController tests', () => {

  beforeEach(() => {
    store.clearActions();
  })

  it('should render the panel', () => {
    const comp = wrapper()
    expect(comp.html()).toMatchSnapshot()
  })

  it('should select a spell', () => {
    const comp = wrapper()

    comp.find('div').last().simulate('click')
    expect(store.getActions()).toEqual([
      {"spell": {"level": "0", "name": "fire bolt"}, "type": "SELECT_SPELL"}
    ])
  })

  it('should deselect a spell', () => {
    const comp = wrapper()

    comp.find('span').first().simulate('click')
    expect(store.getActions()).toEqual( [{"spell": {"level": "0", "name": "friends"}, "type": "DESELECT_SPELL"}])
  })

  it('should clear all spells', () => {
    const comp = wrapper()

    comp.find('button').first().simulate('click')
    expect(store.getActions()).toEqual([ { type: 'CLEAR_SPELL_SELECTION' } ])
  })
})