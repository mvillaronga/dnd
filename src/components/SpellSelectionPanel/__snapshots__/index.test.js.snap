// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpellSelectionPanel tests should render the panel 1`] = `"<div><h1>Spells</h1><hr><div>Filter&nbsp;<input type=\\"text\\"></div><hr><div><button>Clear Selection</button><button>Generate Spell Cards</button><div><span>friends</span></div><hr></div><div><div>magic missle - 1</div><div>find familliar - 1</div><div>fire bolt - 0</div></div></div>"`;
